File: credit.c
#include <cs50.h>
#include <stdio.h>


int main(void)
{

    //required variables
    long number = get_long("Number:"); //get the card number
    int sum = 0, last, second, s1, s2;
    long code = number;
    long holder = number;
    long size = number;
    int counter = 0;

    //count the digits
    while (size > 0)
    {
         size = size / 10;
         counter++;
    }


//perform validation
if (counter != 16 && counter != 15 && counter != 13)
    {
        printf("INVALID\n");
        return 0;
    }

//perform checksum
    do
    {


        last = holder % 10;
        holder = holder / 10;
        second = (holder % 10) * 2;
        s1 = second % 10;
        s2 = second / 10;

        holder = holder / 10;

        sum = sum + last + s1 + s2;
    }
    while (holder > 0);




    //validate and print the company name else "INVALID"
    if (sum % 10 == 0)
    {

        do
        {
            code = code / 10;
        }
        while (code > 100);

        if (code == 34 || code == 37)
        {

            printf("AMEX\n");
        }

        else if (code > 50 && code < 56)
        {
            printf("MASTERCARD\n");
        }

        else if (code / 10 == 4)
        {
            printf("VISA\n");
        }

        else
        {

            printf("INVALID\n");

        }


    }
    else
    {
        printf("INVALID\n");

    }



}

File: credit.py
from cs50 import get_int


def main():
    
    # get the credit card number
    num = get_int("Number: ")
    
    # if the number checksum is valid print the company name
    if check(num):
        companyName(int(str(num)[0:2]))  # passing only the first two digit 
    else:
        print("INVALID\n")


# run the checksum algorithm
def check(num):

    Sum = 0
    
    # reverse and loop through the digits
    for i, n in enumerate(reversed(str(num))):
        
        # add only the last digits 
        if i % 2 == 0:
            Sum += int(n)
        else:
            for j in str(int(n)*2):
                Sum += int(j)

    if Sum % 10 == 0:
        return True
    else:
        return False
        

# determine the company name based on the first two numbers
def companyName(num):

    if (num == 34 or num == 37):
        print("AMEX\n")
    elif (num > 50 and num < 56):
        print("MASTERCARD\n")
    else:
        print("VISA\n")


main()